pkg/cmd/cli/nodeagent/server.go

```diff
@@ -333,6 +337,62 @@ func (s *nodeAgentServer) markInProgressCRsFailed() {
 	s.markInProgressPVRsFailed(client)
 }
 
+func (s *nodeAgentServer) markDataUploadsCancel(client ctrlclient.Client, r *controller.DataUploadReconciler) {
+	pods := &v1.PodList{}
+	if err := client.List(s.ctx, pods); err != nil {
+		s.logger.WithError(errors.WithStack(err)).Error("failed to list pods on current node")
+		return
+	}
+	for _, pod := range pods.Items {
+		du, err := r.FindDataUploadByPod(&pod)
+		if err != nil {
+			s.logger.WithError(errors.WithStack(err)).Errorf("failed to get dataupload by pod %s/%s", pod.Namespace, pod.Name)
+			continue
+		}
+		updated := du.DeepCopy()
+		updated.Spec.Cancel = true
+		du.Status.Phase = velerov2alpha1api.DataUploadPhaseCanceled
+		if du.Status.StartTimestamp.IsZero() {
+			du.Status.StartTimestamp = &metav1.Time{Time: r.Clock.Now()}
+		}
+		du.Status.CompletionTimestamp = &metav1.Time{Time: r.Clock.Now()}
+		updated.Status.Message = fmt.Sprintf("found a dataupload with status %q during the server starting, mark it as cancel", du.Status.Phase)
+		if err := client.Patch(s.ctx, updated, ctrlclient.MergeFrom(du)); err != nil {
+			s.logger.WithError(errors.WithStack(err)).Errorf("failed to mark dataupload %q cancel", du.GetName())
+			continue
+		}
+		s.logger.WithField("dataupload", du.GetName()).Warn(du.Status.Message)
+	}
+}
+
+func (s *nodeAgentServer) markDataDownloadsCancel(client ctrlclient.Client, r *controller.DataDownloadReconciler) {
+	pods := &v1.PodList{}
+	if err := client.List(s.ctx, pods); err != nil {
+		s.logger.WithError(errors.WithStack(err)).Error("failed to list pods on current node")
+		return
+	}
+	for _, pod := range pods.Items {
+		dd, err := r.FindDataDownloadByPod(&pod)
+		if err != nil {
+			s.logger.WithError(errors.WithStack(err)).Errorf("failed to get datadownload by pod %s/%s", pod.Namespace, pod.Name)
+			continue
+		}
+		updated := dd.DeepCopy()
+		updated.Spec.Cancel = true
+		updated.Status.Message = fmt.Sprintf("found a datadownload with status %q during the server starting, mark it as cancel", dd.Status.Phase)
```

Comment:
The same as above, change it to `during node-agent starting`

Reply:
Modified